### PR TITLE
C2LC-196: Limit the character movement to within the grid

### PR DESCRIPTION
**Pull request by simonbates**
This work isn't ready yet but I've created the pull request to make a Netlify deployment for the branch and to share the code so far.

While I believe that the code is behaving as we had planned, I think it actually isn't quite what we want.

For example, if you run the following program:

- left, forward, forward, forward, left, left, forward

The character stops as expected at the edge of the scene. However, when it moves back into the scene, it is now moving from grid line to grid line, rather than from centre of grid cell to centre of grid cell. It is moving one grid cell width at a time, but when it was stopped at the edge of the scene, it became out of phase.

### CODE REVIEW

**Comment by chosww** [comment]
Thanks Simon, your implementation works nicely. About the problem you mentioned, I am thinking maybe we could have a boolean state (isOffBound) to keep track of whether the character is outside of the bound or not, and if it is outside of the bound and the next movement moves the character back to within the bound, it will move the character only half cell width to the grid and change the state respectively.

**Review by chosww — COMMENTED** [review]
Thanks @simonbates for the work; storing dimensions related to the scene in one place, and using properly named methods for getting values for the dimensions made it easier to maintain and understand the source code. I just left a minor suggestion about having a comment or a variable for the constant that represents half of gridCell's width.

File: src/Scene.js [hunk 1]
```diff
@@ -43,31 +43,31 @@ class Scene extends React.Component<SceneProps, {}> {
                     textAnchor='end'
                     key={`grid-cell-label-${i}`}
                     dominantBaseline='middle'
-                    x={minX - rowLabelOffset}
-                    y={yOffset - this.props.gridCellWidth / 2}>
+                    x={this.props.dimensions.getMinX() - rowLabelOffset}
+                    y={yOffset - 0.5}>
```
[the remark this answers is not in the document]
Comment (simonbates) [reply src/Scene.js]: Done

File: src/Scene.js [hunk 2]
```diff
@@ -43,31 +43,31 @@ class Scene extends React.Component<SceneProps, {}> {
...
+                    y2={this.props.dimensions.getMaxY()} />);
             }
             grid.push(
                 <text
                     className='Scene__grid-label'
                     key={`grid-cell-label-${String.fromCharCode(64+i)}`}
                     textAnchor='middle'
-                    x={xOffset - this.props.gridCellWidth / 2}
-                    y={minY - columnLabelOffset}>
+                    x={xOffset - 0.5}
```
[the remark this answers is not in the document]
Comment (simonbates) [reply src/Scene.js]: Done

**Pull request merged**